### ISSUE TITLE
v2: DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead

### ISSUE THREAD

**Issue by brandonkal** [gatsbyjs/gatsby#7884]
## Description

After running gatsby build with Gatsby v2, I now receive this depreciation warning:
```
(node:6712) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
```

### Steps to reproduce

1. `gatsby build`
2. Observe warning after Javascript and CSS bundles are built

### Expected result

No depreciation warning

### Actual result

There is a depreciation warning

### Environment

Run `gatsby info --clipboard` in your project directory and paste the output here. Not working? You may need to update your global gatsby-cli - `npm install -g gatsby-cli`

```
  System:
    OS: macOS Sierra 10.12.6
    CPU: x64 Intel(R) Core(TM) i7-2600K CPU @ 3.40GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 8.11.4 - ~/.nvm/versions/node/v8.11.4/bin/node
    Yarn: 1.9.4 - /usr/local/bin/yarn
    npm: 5.10.0 - ~/.nvm/versions/node/v8.11.4/bin/npm
  Browsers:
    Chrome: 68.0.3440.106
    Firefox: 61.0.2
    Safari: 11.1.2
  npmPackages:
    gatsby: next => 2.0.0-rc.10 
    gatsby-image: next => 2.0.0-rc.1 
    gatsby-plugin-canonical-urls: next => 2.0.0-rc.1 
    gatsby-plugin-feed: next => 2.0.0-rc.2 
    gatsby-plugin-google-analytics: next => 2.0.0-rc.1 
    gatsby-plugin-manifest: next => 2.0.2-rc.1 
    gatsby-plugin-netlify: next => 2.0.0-rc.1 
    gatsby-plugin-nprogress: next => 2.0.0-rc.1 
    gatsby-plugin-offline: next => 2.0.0-rc.2 
    gatsby-plugin-react-helmet: next => 3.0.0-rc.1 
    gatsby-plugin-sitemap: next => 2.0.0-rc.1 
    gatsby-plugin-styled-components: next => 3.0.0-rc.1 
    gatsby-remark-images-contentful: ^1.0.2 => 1.0.2 
    gatsby-remark-prismjs: next => 3.0.0-rc.2 
    gatsby-source-contentful: next => 2.0.1-rc.3 
    gatsby-source-filesystem: next => 2.0.1-rc.1 
    gatsby-transformer-remark: next => 2.1.1-rc.1 
  npmGlobalPackages:
    gatsby-cli: 1.1.58
```

**Comment by kakadiadarpan** [comment]
Hi @brandonkal, this warning is coming because webpack 4 is using a new plugin system and deprecates the previous APIs. As per a [comment](https://github.com/webpack/webpack/issues/6568#issuecomment-377491754) in the issue in webpack repo, you can find out the plugin which is causing this warning by running your build with option `--trace-deprecation`. Can you provide this stack trace to us?

**Comment by kakadiadarpan** [comment]
You will need to run gatsby using node:

```
node --trace-deprecation ./node_modules/.bin/gatsby build 
```

**Comment by kakadiadarpan** [comment]
Fixed in #7894
